### PR TITLE
feat(Slug): Wrote and tested the Slug class

### PR DESCRIPTION
**Pull request by ericadamski**
:100: test coverage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  },
+};
+
+export default class Slug {
+  constructor(slug: string = ''): Slug {
+    this._original = slug;
+    this._normalized = normalize(slug);
+  }
+
+  toString(options: { format: string } = { format: 'pascal' }): string {
```
Comment (frytyler) [inline src/index.js]: Allow plural

File: src/__tests__/slug.spec.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      // Assert
+      expect(result.includes('_')).to.be.false;
+      expect(result).to.be.equal('pull-requests');
+    });
+  });
+
+  describe('creation', () => {
+    const expectedSlug = 'pull_request';
+
+    describe('plurarl', () => {
```
Comment (pvienneau) [inline src/__tests__/slug.spec.js]: Typo

File: src/__tests__/slug.spec.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          .to.be.a('function')
+          .with.lengthOf(1);
+      });
+
+      it('should match singular underscore deliminated version of slug', () => {
+        // Arrange
+        const slug = new Slug('pull-request');
+
+        // Assert
+        expect(slug.equals('pull-requests')).to.be.true;
```
Comment (pvienneau) [inline src/__tests__/slug.spec.js]: Is it forgiving enough for all slug instances to be case insensitive?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import pluralize, { singular } from 'pluralize';
+
+function normalize(slug: string): string {
```
Comment (pvienneau) [inline src/index.js:3]: Curious - does flow play nicely with composed functions, given its argument isn't defined at declaration?

**Review by pvienneau — CHANGES_REQUESTED** [review]
I'm thinking that the camelCasing of a string would be a very common operation to perform (ie getting our current field components, iterating through our redux store through models), IMO it would be cleaner to be able to call `.camelCase()` over `.toString({format: 'camel'}). I don't see this options object grow 
 much; if it were to grow with additional parameters, I would say this would trigger a re-analysis of the purpose of this class.

Thoughts?

That's the only thing I'm unsure of. The rest is good.

File: src/__tests__/slug.spec.js (hunk 3 above)
Comment (ericadamski) [reply src/__tests__/slug.spec.js]: no, good catch.

I suppose you mean passing something like: `pUlL-RequEst` and have that normalized?

File: src/index.js (hunk 4 above)
Comment (ericadamski) [reply src/index.js:3]: I don't really understand the question

**Review by frytyler — APPROVED** [review]

File: src/index.js (hunk 4 above)
Comment (ericadamski) [reply src/index.js:3]: https://github.com/mb3online/expresso/blob/8a2ca65e8bee896fe040c18dc67e1486cd1f6e94/services/model/reduce-relationships.js#L5

**Comment by ericadamski** [comment]
@pvienneau how much do you care about `.camelCase(🐪)`?

**Review by frytyler — APPROVED** [review]

**Review by pvienneau — APPROVED** [review]

**Pull request merged**